### DISCUSSION TITLE
Is there a way to use render_macros recursively

### DISCUSSION THREAD

**Discussion by mhussein** [fralau/mkdocs-macros-plugin#203] [Q&A] [answered]
I have few directories in a very large project that uses macros while the rest doesn't. The project is set as [opt-in](https://mkdocs-macros-plugin.readthedocs.io/en/latest/rendering/#solution-2-opt-in-specify-which-pages-must-be-rendered) with 
```
render_by_default: false
```

Is there a way to avoid setting `render_macros: true` on every single file in those directories that use macros? Is there a way to tell the plugin to run on all files under certain folder(s)?

**Comment by fralau** [comment]
No there isn't, I hadn't thought about it. 

But it's an intriguing idea 🤔  How many files does your MkDocs project have? How many files per directory?

**Reply by mhussein** [reply]
It is hard to know for sure as it is a collection of multiple repositories and I don't have the full resolved set. An estimate would be a folder of ~400 markdown files that I want to enable macros on vs ~2000 files that I don't have control over.

**Reply by fralau** [reply]
Wow, that's a big one 😄 

What was the reason for adopting opt-in? Do you have random files which contain syntax incompatible with MkDocs-Macros?

**Reply by mhussein** [reply]
The opt-in is set on the main repository that collects the documentation from multiple sources. I am not sure why the decision was made, but I believe it is to reduce the impact on different teams working on their docs. One less thing to think about maybe. With the thought that if you need macros it is easy to enable it where needed.

Our set of documents uses macros extensively and we had it enabled in our repo, but we are now integrating our docs in the "main" repositories, and of course none of the macros work unless we add the `render_macros: true` on each single file.

I was trying to see if there is a way to avoid doing that, but it seems there isn't for now. I can try to check with the maintainers of the main repo to see if they can enable it, but I wouldn't if I were in their shoes because of the number of teams/files involved that have to check for impact.

**Reply by fralau** [reply]
I can appreciate both your position and theirs, and now that you mention that you are kind of merging an MkDocs system into another one, it makes sense.

**Reply by fralau** [reply]
I think I will open an issue about this.

**Reply by fralau** [reply, marked as answer]
Opened issue #204

**Reply by mhussein** [reply]
Thanks a lot, appreciate you taking interest in the issue.
